jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/DataUploader.java

```diff
@@ -102,6 +102,10 @@ public static UploadDetails incomingData(HttpAction action, StreamRDF dest) {
         }
     }
 
+    // Jetty requires a setting of this annotation object as a request attribute.
+     private static MultipartConfigElement multipartConfigElement = new MultipartConfigElement("");
+     private static String multipartAttributeName = org.eclipse.jetty.server.Request.__MULTIPART_CONFIG_ELEMENT;
```

Comment:
Do you know if this `__MULTIPART_CONFIG_ELEMENT` attribute was created this way to discourage external use?

Reply:
Hm, can't find any note saying that, so probably just a different way to name constants I guess. https://github.com/eclipse/jetty.project/blob/410e9db837c964bfcb27ff01693eb77a45dcd60f/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java#L144

Reply:
> Do you know if this __MULTIPART_CONFIG_ELEMENT attribute was created this way to discourage external use?

I'll check some more. Using the annotation `@MultipartConfig` on FusekiFilter does not work (it's not a servlet).

The name is Jetty specific which is why it is protected by the `getAttribute`.  Jetty source code likes leading `_`'s. See `org.eclipse.jetty.server.Request`.

I'm unsure if Tomcat needs anything special. I haven't managed to find a mention one way or the other.

PS: https://stackoverflow.com/questions/8047173/how-to-use-httpservletrequestgetparts-in-a-servlet-filter-running-on-tomcat

got me to:

https://tomcat.apache.org/tomcat-7.0-doc/config/context.html#Common_Attributes

Set `allowCasualMultipartParsing`.

Reply:
There is no rush to complete this.

@kinow - thanks for catching the issues - I hadn't noticed that the constant was Jetty-specific.

With various other things doing on, clarifying the situation for Tomcat may not get done for 4.7.0.

There may be a way to automatically programtically  set `allowCasualMultipartParsing` in Tomcat, otherwise the user needs to set it in their `context.xml`.

Marked as "draft" to avoid accidentally merging!